### ISSUE TITLE
implement a interface, not check the return type

### ISSUE THREAD

**Issue by zengwe** [microsoft/TypeScript#35256]
```typescript
interface IHuman {
    run():Promise<number>;
    getAge():number;
}
class person implements IHuman {
    run() {
        return '123';
    }
    getAge() {
        return 18;
    }
}

class person2 implements IHuman {
    run():Promise<number> {
        return 'fadsfads';
    }
    getAge() {
        return 18;
    }
}
```
Compilation does not report any errors.
vscode tips: person2.run:any;

**Comment by DanielRosenwasser** [comment]
I don't see any unexpected behavior. What is the expected behavior that you are not seeing?
